grpc-java/grpc-servlet-jakarta/src/main/java/io/grpc/servlet/jakarta/ServletServerStream.java

```diff
@@ -303,8 +303,13 @@ public void cancel(Status status) {
             close(Status.CANCELLED.withCause(status.asRuntimeException()), new Metadata());
             CountDownLatch countDownLatch = new CountDownLatch(1);
             transportState.runOnTransportThread(() -> {
-                asyncCtx.complete();
-                countDownLatch.countDown();
+                try {
+                    asyncCtx.complete();
+                    countDownLatch.countDown();
+                } catch (final Exception err) {
+                    // this happens when the async context is already complete
+                    logger.info("Unexpected exception completing async context: " + err.getMessage());
```

Comment:
do we want to log this? i see you're not logging for io errors when writing headers/trailers below - those go out in success or failure causes, but this is explicitly the cancel() method, so we are already expecting failure, it is less surprising to see an error here?